site-packages/integralstor/cifs.py

```diff
@@ -531,18 +531,20 @@ def generate_smb_conf():
         return True, None
 
 
-def reload_configuration():
+def reload_configuration(action='restart'):
     try:
+        if action not in ['restart', 'reload']:
+            raise Exception('Error: Received invalid action')
         cmd_to_run = 'smbcontrol all reload-config'
```

Comment:
Hey, aren't we missing the main part of this function or am I missing something?! Shouldn;t the cmd_to_run change based on the action?

Reply:
Call to `services_management.update_service_status()` changes based on action, `cmd_to_run` runs regardless of which action is passed.

Reply:
Since `services_management.update_service_status()` also reloads the config, Should `cmd_to_run` be removed?